### PR TITLE
File store backup

### PR DESCRIPTION
**Pull request by manuroe**
https://github.com/vector-im/vector-ios/issues/376

### CODE REVIEW

File: MatrixSDK/Data/Store/MXFileStore/MXFileStore.m [hunk 1]
```diff
@@ -48,9 +49,14 @@ @interface MXFileStore ()
...
+    NSMutableArray *roomsToCommitForDeletion;
+
     // The path of the MXFileStore folder
     NSString *storePath;
 
+    // The path of the backup folder
+    NSString *storeBackupPath;
+
     // The path of the temporary file created during saving process.
     NSString *savingMarkerFile;
```
Comment (giomfo) [inline MatrixSDK/Data/Store/MXFileStore/MXFileStore.m]: This parameter 'savingMarkerFile ' is not used anymore

File: MatrixSDK/Data/Store/MXFileStore/MXFileStore.m [hunk 2]
```diff
@@ -496,57 +506,143 @@ - (MXMemoryRoomStore*)getOrCreateRoomStore:(NSString*)roomId
...
-    NSString *roomFolder = [self folderForRoom:roomId];
+    NSString *storeBackupRoomsPath;
+
+    if (backupEventStreamToken)
+    {
+        storeBackupRoomsPath = [[storeBackupPath stringByAppendingPathComponent:backupEventStreamToken]
+                                stringByAppendingPathComponent:kMXFileStoreRoomsFolder];
+    }
+
+    return backupEventStreamToken;
```
Comment (giomfo) [inline MatrixSDK/Data/Store/MXFileStore/MXFileStore.m]: The wrong string is returned here, it should be 'storeBackupRoomsPath'.

File: MatrixSDK/Data/Store/MXFileStore/MXFileStore.m [hunk 3]
```diff
@@ -496,57 +506,143 @@ - (MXMemoryRoomStore*)getOrCreateRoomStore:(NSString*)roomId
...
+- (NSString*)metaDataFileForBackup:(BOOL)backup
 {
-    return [[storeRoomsPath stringByAppendingPathComponent:roomId] stringByAppendingPathComponent:kMXFileStoreRoomReadReceiptsFile];
+    if (!backup)
+    {
+        return [storePath stringByAppendingPathComponent:kMXFileStoreMedaDataFile];
+    }
+    else
+    {
+        return [[storeBackupPath stringByAppendingPathComponent:backupEventStreamToken] stringByAppendingPathComponent:kMXFileStoreMedaDataFile];
```
Comment (giomfo) [inline MatrixSDK/Data/Store/MXFileStore/MXFileStore.m]: backupEventStreamToken may be nil here (see the MetaData saving at the first login)

Comment (manuroe) [inline MatrixSDK/Data/Store/MXFileStore/MXFileStore.m]: It is not a big issue as the value returned by this method is used only if there is a file to backup. On the first login, there is not yet a MetaData file so no issue.

**Pull request merged**